ars/core/repository.py

```diff
@@ -0,0 +1,187 @@
+            raise self.AccessRequestInvalidDuration("Access start date is invalid")
+        access_ends = creation_data.access_ends
+        if not self._min_duration <= access_ends - access_starts <= self._max_duration:
+            raise self.AccessRequestInvalidDuration("Access end date is invalid")
+
+        full_user_name = auth_context.name
+        if auth_context.title:
+            full_user_name = auth_context.title + " " + full_user_name
+
+        access_request_data = AccessRequestData(
+            **creation_data.dict(),
+            full_user_name=full_user_name,
+            request_created=request_created,
+        )
+
+        return await self._dao.insert(access_request_data)
+
+    async def get(
+        self,
+        *,
+        dataset_id: Optional[str] = None,
+        user_id: Optional[str] = None,
+        status: Optional[AccessRequestStatus] = None,
+        auth_context: AuthContext,
+    ) -> list[AccessRequest]:
+        """Get the list of all access requests with the given properties.
+
+        Only data stewards may list requests created by other users.
+
+        Raises an AccessRequestError if the user is not authorized.
+        """
+
+        if not auth_context.id:
+            raise self.AccessRequestError("Not authorized")
+        if not has_role(auth_context, DATA_STEWARD_ROLE):
+            if user_id is None:
+                user_id = auth_context.id
+            elif user_id != auth_context.id:
+                raise self.AccessRequestError("Not authorized")
+
+        mapping: dict[str, Any] = {}
+        if user_id is not None:
+            mapping["user_id"] = user_id
+        if dataset_id is not None:
+            mapping["dataset_id"] = dataset_id
+        if status is not None:
+            mapping["status"] = status
+
+        return [request async for request in self._dao.find_all(mapping=mapping)]
+
+    async def update(
+        self,
+        access_request_id: str,
+        *,
+        status: AccessRequestStatus,
+        auth_context: AuthContext,
+    ) -> None:
+        """Update the status of the access request.
+
+        Only data stewards may use this method.
+
+        Raises an AccessRequestError if the user is not authorized.
+        """
+
+        user_id = auth_context.id
+        if not user_id or not has_role(auth_context, DATA_STEWARD_ROLE):
+            raise self.AccessRequestError("Not authorized")
+
+        try:
+            request = await self._dao.get_by_id(access_request_id)
+        except ResourceNotFoundError as error:
+            raise self.AccessRequestNotFoundError("Access request not found") from error
+        if request.status == status:
+            raise self.AccessRequestError("Same status is already set")
+        if request.status != AccessRequestStatus.PENDING:
+            raise self.AccessRequestError("Status cannot be reverted")
+
+        # Should set the status in the claims repository here
+        # and proceed only if this succeeds.
```

Comment:
We submit a claim if approved and do nothing if rejected, right?

Reply:
Exactly.